guides/kubernetes.md

```diff
@@ -0,0 +1,136 @@
+
+## Probes
+Kubernetes has two types of probes, readiness and liveness.
+
+Kubernetes uses readiness probes to know when a container is ready to start accepting traffic.
+
+Kubernetes uses liveness probes to know when to restart a container.
+
+### Configuring probes
+Probes can be configured in the Helm chart on a `Deployment` resource, under the container node.
+
+The above is a simple example of an Http readiness and liveness probes.
+
+```
+readinessProbe:
+  path: /healthy
+  port: 3000
+  initialDelaySeconds: 10
+  periodSeconds: 10
+  failureThreshold: 3
+
+livenessProbe:
+  path: /healthz
+  port: 3000
+  initialDelaySeconds: 10
+  periodSeconds: 10
+  failureThreshold: 3
+```
+
+In this example, the cluster will wait for 10 seconds after the pod is deployed.  It will then poll both the liveness and readiness endpoints on port 3000 every 10 seconds.  
+
+If it receives three successive status codes other than 200 for the readiness probe it will stop routing traffic to that pod.
+
+If it receives three successive status codes other than 200 for the liveness probe it will assume the pod is unresponsive and kill it.
+
+**Note** that a liveness probe works in conjunction with the restartPolicy value. In order to restart the restartPolicy must be set to Always or OnFailure.
+
+#### Values
+`path`: the URL route the liveness probe should sent a response to.
+
+`port`: the port on which the service is exposing
+
+`initialDelaySeconds`: how long before the first probe should be sent. This should be safely longer than it takes the pod to start up, otherwise the pod could be stuck in a reboot loop
+
+`periodSeconds`: how often the liveness probe should check the pod is responsive. Recommendation is between 10 and 20 seconds
+
+`failureThreshold`: how many probe failures before the pod is automatically restarted
+
+As well as Http probes, there are also command and TCP based probes, full details can be found in the documentation https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/
+
+## Descheduler
+Scheduling in Kubernetes is the process of binding pending pods to nodes, and is performed by a component of Kubernetes called kube-scheduler. The scheduler's decisions, whether or where a pod can or can not be scheduled, are guided by its configurable policy which comprises of set of rules, called predicates and priorities. The scheduler's decisions are influenced by its view of a Kubernetes cluster at that point of time when a new pod appears for scheduling. As Kubernetes clusters are very dynamic and their state changes over time, there may be desire to move already running pods to some other nodes for various reasons:
+
+Some nodes are under or over utilized.
+The original scheduling decision does not hold true any more, as taints or labels are added to or removed from nodes, pod/node affinity requirements are not satisfied any more.
```

Comment:
This section doesn't render too well as all the lines becomes appended, rather than displaying each reason on a separate line.

 It might be more readable if bullet points are used for each reason.